src/merge_outputs.rs

```diff
@@ -26,7 +26,7 @@ pub struct LayerUploads {
 pub async fn merge(
     merge_config: &super::merge_config::MergeConfig,
     relative_search_path: &Option<PathBuf>,
-    external_execution_config: &Option<ExecutionConfig>,
+    external_execution_configs: &Vec<ExecutionConfig>,
```

Comment:
nit: I if you remove the `&Vec` here and just do `Vec<ExecutionConfig>` we can avoid the clone below. Seems like the caller is the last to use this config.